src/python/py/models/quantized_model.py

```diff
@@ -339,10 +351,39 @@ def __init__(self, quant_type, input_path, bits, group_size, q_size, kv_size, in
         # Set properties of each layer based on quantization type
         self.set_properties()
 
+    def _initialize_quantized_lm_head(self, bits, group_size):
+        """
+        Initialize `QuantizedTensorModule` for LM head if not already set
+        """
```

Comment:
Should we consider replacing `TensorModule` with `QuantizedTensorModule`, moving `self.weight` and `self.bias` to `QuantizedTensorModule`, and just use `QuantizedTensorModule`? Then, we wouldn't need to initialize and we can still check between quantized LM head vs. unquantized LM head with `self.lm.qweight is not None`.